### DISCUSSION TITLE
URL 포트번호

### DISCUSSION THREAD

**Discussion by sa02045** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#19] [Interview Question]
URL의 포트는 어떤 역할을 하는가? 포트번호가 다르다는 것은 어떤 의미일까? CORS 정책에서 왜 포트번호가 다르면 다른 Origin(출처)로 인식할까?

**Comment by Hyevvy** [comment]
URL 의 포트번호는 서버가 열어놓은 네트워크 포트를 의미하며, HTTP 의 기본 포트는 80이다.
포트는 컴퓨터안에서 프로그램을 찾을 때 사용된다. 그러므로 포트번호가 다르다는 것은 다른 프로그램이라는 의미다. 
CORS 정책은 다른 출처간의 리소스 접근, 공유 권한을 처리하는 보안 정책인데, 이 때 같은 origin은 프로토콜, 호스트, 포트 번호가 같은 경우를 의미한다. 포트 번호가 다른 경우는 컴퓨터 안에서 다른 프로그램을 의미하므로 다른 출처로 인식한다.
예를 들어, localhost:3000 으로는 react프로젝트를 띄우고, localhost:3001 로는 vue 프로젝트를 띄우는 것과 같다. 포트번호가 다른 것 하나만으로도 다른 프로젝트라는 뜻이다.

**Reply by punchdrunkard** [reply]
이것보다 좋은 답변은 생각나지 않네요ㅎㅎ.......
